### ISSUE TITLE
Camera CFG and dropout

### ISSUE THREAD

**Issue by stalkerrush** [snap-research/ac3d#23]
Hi @sherwinbahmani , thanks for releasing the great work. In the paper, it was mentioned that classifier-free guidance on camera and a camera dropout of 10% (training-time) was used. However, in the code, I wasn't able to find relevant lines. Are these not included in the release? If I want to add the camera CFG, do I change the first half of this [controlnet_latents](https://github.com/snap-research/ac3d/blob/3c1e29e688f4a6d0f0ad41f1bf75d2eab709dac2/controlnet_pipeline.py#L579) to be zeros (e.g. torch.cat([torch.zeros_like(controlnet_latents), controlnet_latents]))? 

Thanks!

**Comment by sherwinbahmani** [comment]
Hi,

You are correct, this open-source implementation does not have the camera cfg and dropout included.
What you are looking at is the text cfg. You would need another forward pass where the camera is dropped and during training you also have to zero-out the plucker embedding with a given probability. Hence, in total there are 3 forward passes. So you need something like this:

`controlnet_latents = torch.cat([controlnet_latents] * 3)`

One takes text and camera, one only text, and one only camera. To fuse the forward passes, you then have to follow the equation below:

<img width="423" alt="Image" src="https://github.com/user-attachments/assets/73e87b2c-a228-4002-ab5b-f8bdda2398c0" />

**Comment by stalkerrush** [comment]
I see, thanks for the reply. Is this open source checkpoint trained to support zero plucker embedding already?
